### ISSUE TITLE
howard-hinnant-date: CMake configuration points to `out` instead of `dev` for includes

### ISSUE THREAD

**Issue by jktjkt** [NixOS/nixpkgs#163212]
### Describe the bug
I'm trying to use `howard-hinnant-date` from a CMake project. Its build fails because the include (`.h`) files are under a prefix which is not referenced from the installed CMake config file. Removing the `outputs = [ "out" "dev" ];` fixes this bug.

### Steps To Reproduce
1. download [`CMakeLists.txt`](https://github.com/NixOS/nixpkgs/files/8201196/CMakeLists.txt) and [`yay.cpp` (might need removing the trailing `.txt` on github)](https://github.com/NixOS/nixpkgs/files/8201201/yay.cpp.txt)
2. store them into, say, `~/temp/nix-howard-date`
3. get a `nix-shell -p cmake howard-hinnant-date`
4. try to build via `cmake ~/temp/nix-howard-date`

Here's how it fails on my NixOS:
```console-session
$ nix-shell -p cmake howard-hinnant-date
$ rm -rf xx; mkdir xx; cd xx; cmake ~/temp/nix-howard-date
-- The CXX compiler identification is GNU 10.3.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/pja9g36cy32z3d51942jqk91a6l2d5nv-gcc-wrapper-10.3.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Configuring done
CMake Error at CMakeLists.txt:5 (add_executable):
  Cannot find source file:

    /nix/store/wjkxpmp3x7rg7zl6apsl6v4h6pjif3sr-howard-hinnant-date-3.0.1/include/date/date.h

  Tried extensions .c .C .c++ .cc .cpp .cxx .cu .mpp .m .M .mm .ixx .cppm .h
  .hh .h++ .hm .hpp .hxx .in .txx .f .F .for .f77 .f90 .f95 .f03 .hip .ispc


CMake Error in CMakeLists.txt:
  Imported target "date::date" includes non-existent path

    "/nix/store/wjkxpmp3x7rg7zl6apsl6v4h6pjif3sr-howard-hinnant-date-3.0.1/include"

  in its INTERFACE_INCLUDE_DIRECTORIES.  Possible reasons include:

  * The path was deleted, renamed, or moved to another location.

  * An install or uninstall procedure did not complete successfully.

  * The installation package was faulty and references files it does not
  provide.



-- Generating done
CMake Generate step failed.  Build files cannot be regenerated correctly.
```

I *think* that this is similar to, e.g., #97534 and that not splitting the output into several packages will help.

### Notify maintainers
maintainer: @r-burns 
cc: @jtojnar (is the [upstream package](https://github.com/HowardHinnant/date) using CMake properly?)

### Metadata

```console
$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.24, NixOS, 22.05 (Quokka)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.6.1`
 - channels(root): `"nixos, nixos-21.11-21.11"`
 - channels(jkt): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by jtojnar** [comment]
`$<INSTALL_INTERFACE:include>` should be `$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>` in https://github.com/HowardHinnant/date/blob/9ea5654c1206e19245dc21d8a2c433e090c8c3f5/CMakeLists.txt#L66

**Comment by esp32wrangler** [comment]
https://github.com/HowardHinnant/date/blob/9ea5654c1206e19245dc21d8a2c433e090c8c3f5/CMakeLists.txt#L122 as well

**Comment by panicgh** [comment]
This was probably fixed by #199323. The example passes on NixOS 23.05, x86_64-linux:

```console
[user@host:~/test/uu]$ nix-shell -p cmake howard-hinnant-date
[nix-shell:~/test/uu]$ mkdir build && cd build
[nix-shell:~/test/build]$ cmake ..
-- The CXX compiler identification is GNU 12.2.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /nix/store/lcf37pgp3rgww67v9x2990hbfwx96c1w-gcc-wrapper-12.2.0/bin/g++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Configuring done
-- Generating done
-- Build files have been written to: ~/test/build
[nix-shell:~/test/build]$ make
[ 50%] Building CXX object CMakeFiles/yay.dir/yay.cpp.o
[100%] Linking CXX executable yay
[100%] Built target yay
[nix-shell:~/test/build]$ ./yay
[nix-shell:~/test/build]$ echo $?
0
```
